src/Analyzers/CSharp/Analyzers/UseUTF8StringLiteral/UseUTF8StringLiteralDiagnosticAnalyzer.cs

```diff
@@ -87,8 +95,21 @@ private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol
 
             var properties = ImmutableDictionary<string, string?>.Empty.Add(StringValuePropertyName, stringValue);
 
+            var location = arrayCreationExpression.Syntax.GetFirstToken().GetLocation();
+
+            // Special case - for a param array, the syntax is the entire invocation, because there is no one
+            // syntax node for just the array elements, so we construct our own location. The code fix has
+            // special handling for this too.
+            if (arrayCreationExpression.Syntax is InvocationExpressionSyntax)
```

Comment:
Sorry for an edge case again 😄

If the `params` parameter is in constructor, the caller will be `BaseObjectCreationExpressionSyntax`

Reply:
Another one 😢 

```
public class C {
    public int this[params byte[] bytes]
    {
        get => 0;
    }

    public void M()
    {
        _ = this[0, 1, 2];
    }
}
```

Reply:
Also `PrimaryConstructorBaseTypeSyntax` and `ConstructorInitializerSyntax`

Reply:
could you just look for the containing BaseARgumentList?